### PR TITLE
AWS Migration

### PR DESCRIPTION
**Pull request by toadzky**
Switches to gremlin for the backend, adds cloudformation, and some github actions.

### CODE REVIEW

**Comment by ndrwksr** [comment]
I am clueless about anything here but the Kotlin... That said, the Kotlin looks good to me!

**Review by ndrwksr — APPROVED** [review]

**Review by yonglid — CHANGES_REQUESTED** [review]
Already left Slack comments but just so we have it recorded here (and it doesn't get lost) 
- updated changes made by David (custom made data classes like DeviceId need to be unwrapped ref by value when referenced by graph api) 
- talked about adjusting gremlin settings for vertex indices so they aren't restricted to just Longs currently -- issue when doing /users POST and trying to pass a uuid or uuid in string format. 

Only tested endpoints in /users so far and those were the issues I had run into. 

Other thing is I run docker on windows so I usually manually change the config.yml's db host to my default ip address to connect to the db. Unsure if we want to add to the config to account for different developers on different OS, but just adding this in here / also seeing if anyone has advice or suggestions on that

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+echo "*** Startup $0 suceeded now starting service using eval to expand CMD variables ***"
+exec java -jar ${JAVA_OPTS} /app.jar server /config.yml
```
Comment (mbuchel) [inline entrypoint.sh:4]: inconsistent, says eval but uses exec, suggest to switch to `eval "java -jar ${JAVA_OPTS} /app.jar server /config.yml"`

File: pom.xml [hunk 2]
```diff
@@ -95,18 +139,42 @@
             <artifactId>jackson-annotations</artifactId>
         </dependency>
 
-        <!-- libphonenumber -->
+        <!-- Database Dependencies -->
         <dependency>
-            <groupId>com.googlecode.libphonenumber</groupId>
-            <artifactId>libphonenumber</artifactId>
```
Comment (mbuchel) [inline pom.xml:101]: why are we no longer using libphonenumber?

File: src/main/kotlin/io/zerobase/smarttracing/Main.kt [hunk 3]
```diff
@@ -1,28 +1,28 @@
 package io.zerobase.smarttracing
 
-import com.google.i18n.phonenumbers.PhoneNumberUtil
 import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.google.i18n.phonenumbers.PhoneNumberUtil
```
Comment (mbuchel) [inline src/main/kotlin/io/zerobase/smarttracing/Main.kt:4]: is this not from libphonenumber?

File: pom.xml (hunk 2 above)
Comment (toadzky) [reply pom.xml:101]: it just moved down to line 175, under "Other Dependencies". no need to have a special section for single dependencies. i do the grouping so when there are multiple related dependencies (eg aws, dropwizard, gremlin), they are co-located in the file. libphonenumber doesn't have any other pieces to co-locate, so it doesn't need a separate section.

**Comment by toadzky** [comment]
@yonglid i think i fixed the issues with the vertexes and updated the readme accordingly. as for the different ip address thing, making it `localhost` is the most common way to use it. you should be exposing docker container ports for the database - which should make them available on localhost. that being said, if we want to leave the field as a property so it can be overridden with an environment variable in case someone can't use `localhost`, that makes sense.

File: entrypoint.sh (hunk 1 above)
Comment (toadzky) [reply entrypoint.sh:4]: we want `exec` here so that container signals are passed down to the java process.

**Review by yonglid — APPROVED** [review]
we can work on api routes later like you mentioned and I'm not super familiar with the deploy configurations and not familiar with branch-build yml for github actions (?); got it compiled though

**Pull request merged**